citizennet/purescript-ocelot
# Add modal partials

## What does this pull request do?

This PR adds partials so that modals can be closed with the `Escape` key or by clicking outside the modal body.

Partials are a new hybrid between blocks and components that we can leverage for things that don't quite deserve their own component, but still have behaviors that need to run in `eval` to function properly. Partials should be styled just like blocks, except they additionally accept a query they can attach in the right places and provide a block of code that can be run in `HalogenM` to evaluate that query when triggered.

Check out the `initializeModal` and `ifClose` functions for examples.

## Where should the reviewer start?

Review the new modal partial file and the changes to the modal example.

## How should this be manually tested?

Verify that you can open a modal, click Cancel to close it, click outside the body to close it, or type Escape to close it.
